js/handlers.js

```diff
@@ -29,6 +29,15 @@ weechat.factory('handlers', ['$rootScope', '$log', 'models', 'plugins', 'notific
         new_date.setHours(0, 0, 0, 0);
         // Check if the date changed
         if (old_date.valueOf() !== new_date.valueOf()) {
+            var supports_formatting_date = false;
+            var should_be_tuesday = new Date(Date.UTC(2012,11,12,3,12,0))
+                .toLocaleDateString("en-US-u-ca-iso8601", {"weekday":"long"});
```

Comment:
I don't understand at all what you mean by UTC mixup and DST. UTC doesn't have DST. And "mixups" can't happen because UTC is UTC "everywhere". Which means code is just wrong, the way I understand it.

As for init, somewhere in WeechatCtrl, there's already many types of detecting stuff there.